source/common/router/config_impl.cc

```diff
@@ -353,6 +355,65 @@ std::vector<InternalRedirectPredicateSharedPtr> InternalRedirectPolicyImpl::pred
   }
   return predicates;
 }
+PathRewritePolicyImpl::PathRewritePolicyImpl() : enabled_(false){};
+
+PathRewritePolicyImpl::PathRewritePolicyImpl(
+    const envoy::config::core::v3::TypedExtensionConfig typed_config,
+    ProtobufMessage::ValidationVisitor& validator, std::string route_url)
+    : route_url_(route_url), enabled_(true) {
+
+  predicate_factory_ =
```

Comment:
This could be counter to how other extensions store factories if we changed to reference.

Reply:
We generally avoid converting references to pointers unless we need to support them being null, which this never can be.

I think in this case, now that the validation is handled during creation, is it even necessary to store the factory? (That said, I may give you a reason to keep it around in a moment.)

Reply:
I don't see a real need. I've removed storing it. 

I've keep the pointer bc otherwise there are type issues. This should match other extension factories.